src/sql/workbench/contrib/notebook/browser/notebookExplorer/notebookSearchResultsView.ts

```diff
@@ -0,0 +1,378 @@
+		DOM.toggleClass(templateData.lineNumber, 'show', (numLines > 0) || showLineNumbers);
+
+		templateData.lineNumber.textContent = lineNumberStr + extraLinesStr;
+		templateData.lineNumber.setAttribute('title', this.getMatchTitle(match, showLineNumbers));
+
+		templateData.actions.clear();
+		if (this.searchModel.isReplaceActive()) {
+		} else {
+			templateData.actions.push([new RemoveAction(this.searchView.getControl(), match)], { icon: true, label: false });
+		}
+	}
+
+	disposeElement(element: ITreeNode<Match, any>, index: number, templateData: IMatchTemplate): void {
+	}
+
+	disposeTemplate(templateData: IMatchTemplate): void {
+		templateData.actions.dispose();
+	}
+
+	private getMatchTitle(match: Match, showLineNumbers: boolean): string {
+		const startLine = match.range().startLineNumber;
+		const numLines = match.range().endLineNumber - match.range().startLineNumber;
+
+		const lineNumStr = showLineNumbers ?
+			nls.localize('lineNumStr', "From line {0}", startLine, numLines) + ' ' :
+			'';
+
+		const numLinesStr = numLines > 0 ?
+			'+ ' + nls.localize('numLinesStr', "{0} more lines", numLines) :
+			'';
+
+		return lineNumStr + numLinesStr;
+	}
+}
+
+export class SearchAccessibilityProvider implements IListAccessibilityProvider<RenderableMatch> {
+
+	constructor(
+		private searchModel: SearchModel,
+		@ILabelService private readonly labelService: ILabelService
+	) {
+	}
+
+	getWidgetAriaLabel(): string {
+		return nls.localize('search', "Search");
+	}
+
+	getAriaLabel(element: RenderableMatch): string | null {
+		if (element instanceof FolderMatch) {
+			return element.resource ?
+				nls.localize('folderMatchAriaLabel', "{0} matches in folder root {1}, Search result", element.count(), element.name()) :
+				nls.localize('otherFilesAriaLabel', "{0} matches outside of the workspace, Search result", element.count());
+		}
+
+		if (element instanceof FileMatch) {
+			const path = this.labelService.getUriLabel(element.resource, { relative: true }) || element.resource.fsPath;
+
+			return nls.localize('fileMatchAriaLabel', "{0} matches in file {1} of folder {2}, Search result", element.count(), element.name(), paths.dirname(path));
+		}
+
+		if (element instanceof Match) {
+			const match = <Match>element;
+			const searchModel: SearchModel = this.searchModel;
+			const replace = searchModel.isReplaceActive() && !!searchModel.replaceString;
+			const matchString = match.getMatchString();
+			const range = match.range();
+			const matchText = match.text().substr(0, range.endColumn + 150);
+			if (replace) {
+				return nls.localize('replacePreviewResultAria', "Replace '{0}' with '{1}' at column {2} in line {3}", matchString, match.replaceString, range.startColumn + 1, matchText);
+			}
+
+			return nls.localize('searchResultAria', "Found '{0}' at column {1} in line '{2}'", matchString, range.startColumn + 1, matchText);
+		}
+		return null;
+	}
+}
+
+export class SearchDND implements ITreeDragAndDrop<RenderableMatch> {
```

Comment:
This looks really similar to the SearchDND class in searchResultsView.ts. Can we reference that class instead?